kv: slice bounds out of range on kv.writePrimToBuf

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below.

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- Cayley Forum - https://discourse.cayley.io
- IRC - irc.freenode.net #cayley channel

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the command below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

Opening the `bolt2` QuadStore causes a panic due to slice bounds out of range.
This is due to the bloom filter buffer not being initialised before it is used in `writePrimToBuf`.

**Steps to reproduce the issue:**
1. Run this script a couple of times

```
package main

import (
	"github.com/cayleygraph/cayley"
	"github.com/cayleygraph/cayley/graph"
	"github.com/cayleygraph/cayley/quad"

	_ "github.com/cayleygraph/cayley/graph/bolt2"
)

func main() {
	graph.InitQuadStore("bolt2", "db", nil)
	h, _ := cayley.NewGraph("bolt2", "db", nil)

	h.AddQuad(quad.Quad{
		Subject:   quad.IRI("src"),
		Predicate: quad.IRI("x"),
		Object:    quad.IRI("dst"),
	})
}
```

**Received results:**

<!--
Describe results you received.

Any additional information such as logs will help.
Enclose it into following markdown:
-->
```
2017/08/01 14:30:08 Using a bloom filter of 77880040 bytes
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/cayleygraph/cayley/graph/kv.writePrimToBuf(...)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/kv/indexing.go:551
github.com/cayleygraph/cayley/graph/kv.(*QuadStore).initBloomFilter.func1.1(0x226e2e3, 0x8, 0x8, 0x226e2eb, 0x12, 0x12, 0x12, 0x0)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/kv/indexing.go:522 +0x1f7
github.com/boltdb/bolt.(*Bucket).ForEach(0xc4200954c0, 0xc420088560, 0x1, 0xc420088560)
	/Users/jkingsbury/go/src/github.com/boltdb/bolt/bucket.go:390 +0x124
github.com/cayleygraph/cayley/graph/bolt2.Bucket.ForEach(0xc4200954c0, 0xc420088560, 0x3, 0x3)
	<autogenerated>:1 +0x3e
github.com/cayleygraph/cayley/graph/kv.(*QuadStore).initBloomFilter.func1(0x130f2e0, 0xc4200ce0e0, 0xc4200ce0e0, 0x0)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/kv/indexing.go:511 +0xee
github.com/cayleygraph/cayley/graph/kv.View(0x130f2a0, 0xc420098538, 0xc424b57cf0, 0x1, 0x1)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/kv/quadstore.go:80 +0x7f
github.com/cayleygraph/cayley/graph/kv.(*QuadStore).initBloomFilter(0xc4200b6b00, 0x0, 0x0)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/kv/indexing.go:508 +0x210
github.com/cayleygraph/cayley/graph/kv.New(0x130f2a0, 0xc420098538, 0x0, 0x1340000, 0x0, 0x0, 0x0)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/kv/quadstore.go:131 +0x211
github.com/cayleygraph/cayley/graph/bolt2.newQuadStore(0x120dc58, 0x2, 0x0, 0x5, 0xc42008aa68, 0x130c701, 0xc42007c4f0)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/bolt2/bolt.go:74 +0x1fe
github.com/cayleygraph/cayley/graph.NewQuadStore(0x120e022, 0x5, 0x120dc58, 0x2, 0x0, 0xc420053ef8, 0xc420053ec8, 0x10f69fa, 0x120dc58)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/graph/registry.go:56 +0x84
github.com/cayleygraph/cayley.NewGraph(0x120e022, 0x5, 0x120dc58, 0x2, 0x0, 0x130c720, 0xc42007c4f0, 0xc420053f70)
	/Users/jkingsbury/go/src/github.com/cayleygraph/cayley/imports.go:38 +0x66
```

**Expected results:**

The script should not panic

**Output of `cayley version` or commit hash:**

f4d9bdbb47d9c2df37a15dc13d24fb079e3d1255

**Environment details:**

Backend database: `bolt2`

--- a/graph/kv/quadstore.go
+++ b/graph/kv/quadstore.go
@@ -128,9 +128,9 @@ func New(kv KV, _ graph.Options) (graph.QuadStore, error) {
 		return nil, errors.New("bolt: data version is out of date. Run cayleyupgrade for your config to update the data.")
 	}
 	qs.valueLRU = lru.New(2000)
+	qs.bloomBuf = make([]byte, 3*8)
 	qs.initBloomFilter()
 	qs.hashBuf = make([]byte, quad.HashSize)
-	qs.bloomBuf = make([]byte, 3*8)
 	return qs, nil
 }
 

